docs/index.md

```diff
@@ -233,21 +233,45 @@ Note that you can track the progress of mock using the logs stored in `/var/lib/
 
 ## Mock inside Podman, Fedora Toolbox or Docker container
 
-By default, Mock uses [systemd-nspawn](https://www.freedesktop.org/software/systemd/man/systemd-nspawn.html) to isolate the build in chroot.  This is
-not necessary though if you run Mock inside a container, and Mock is the only
-service running there.  NB spawning **containers inside containers** isn't
-implemented in Mock, so switching to `--isolation=simple` is mandatory.  Mock
-is, though, able to automatically detect a container environment, and switch to
-`--isolation=simple`.
-
-One can even run Mock in a rootless Podman container without any special tweaks - the only necessary step is to run the
-container with `--privileged` option.  Read the podman-run manual page for more
-info, but `--privileged` - by the Podman nature - can not give the process more
-permissions than the UID running the podman process already has; in other
-words - `podman run --privileged` is a completely different thing from
-`docker run --privileged`!
-
-So simply, as any **non-privileged user**, do:
+First, we need to state that Mock, in a nutshell, is a tool that (a) prepares an
+appropriate RPM build environment (aka "build chroot" or "buildroot"), and then
+it (b) just executes the RPM build inside (`rpmbuild` run).
+
+The build environment is (again a bit simplified) defined by a set of RPM
+packages that need to be installed in such environment (build dependencies, or
+also `BuildRequires:` in RPM spec files).
+
+Mock is a generic tool to build **any** RPM out there.  And each RPM has a
+different set of requirements (so we can not just pre-generate one environment
+for all packages and share it).  Here comes the important implication:  If you
+want to **build an RPM**, you need to install **other RPMs**, and for that
+you need to have the have **a root access**.
+
+Normally, Mock uses `dnf --installroot /some/directory install ...` (on host)
+to prepare the environment.  Then it switches "inside" the environment using the
+[systemd-nspawn](https://www.freedesktop.org/software/systemd/man/systemd-nspawn.html)
+container (default `--isolation=nspawn`, but you can fallback to `--isolation=simple`
+which is just `man (2) chroot`).  Again [simplified a bit](Feature-bootstrap).
+
+The build itself (`rpmbuild` process) is a non-root operation, Mock
+intentionally drops the privileges there.
+
+All that said, using Mock inside a container to build RPMs is totally possible!
+You just need to have permissions to install RPMs, and be able to switch UID
+when needed (from `root` to non-privileged and back).  As a benefit, we don't
+need to run `systemd-nspawn` and we still have a better isolation (because even
+the `dnf --installroot` is executed inside the container)!
```

Comment:
I am afraid it does not help much if an authenticated daemon (the user of `mock`) runs in the same container as `mock` itself.

Reply:
Right, there's some level of [inner isolation](https://github.com/rpm-software-management/mock/blob/5cc74910a2d1a04b0b820bed3cdb06441d96b214/mock/py/mock.py#L640-L644) but that's rather a historical code we did not touch for a long time (from times when Podman/Docker did not exist, hard to implement right).  It seems more tempting to have one fresh and clean container for each build.  I'll elaborate on this a bit in the docs.  Thanks for the review!